go.mod

```diff
@@ -1,9 +1,10 @@
 module github.com/jhump/protoreflect
 
+go 1.12
```

Comment:
This is correct, before you ask :-) This is what is expected to be in every `go.mod`, it should have been added before (and is added automatically now).

Reply:
The Go tool has yet to automatically add this to any of my `go.mod` files, even when I run `go mod ...` commands. Why do you say it's correct? Do you have a link?

I'm a little concerned about accepting this because it's obvious you are using a different version of Go. Which means this will just result in diff cheese when I go to update things later.

Reply:
Go 1.12 has been out for a while, are you still running Go 1.11?

This is what all new go.mods contain now as of Go 1.12, and is backwards compatible to anything using Go 1.11. It provides extra information to the compiler:

https://golang.org/doc/go1.12#modules

> The go directive in a go.mod file now indicates the version of the language used by the files within that module. It will be set to the current release (go 1.12) if no existing version is present. If the go directive for a module specifies a version newer than the toolchain in use, the go command will attempt to build the packages regardless, and will note the mismatch only if that build fails.

Reply:
> Go 1.12 has been out for a while, are you still running Go 1.11?

Yep. To make sure any runtime/security bugs are shaken out we usually wait a cycle. We will likely update to 1.12 this summer, right before 1.13 is released. (In fact, we have some stuff that we're still working to move off of App Engine, which is sadly stuck on Go 1.9!).

Anyhow, the tests are failing in CI for Go 1.11, so I'm a little skeptical.

Reply:
Again, instead of closing this PR and the go.mod file remaining in an inconsistent state (which it is, it needs cleaning), we could have just removed the `go 1.12` declaration, gotten this merged, and this would be fixed.